std/conv.d

```diff
@@ -875,11 +875,19 @@ T toImpl(T, S)(S value)
         ()@trusted{ memcpy(result.ptr, value.ptr, value.length); }();
         return cast(T) result;
     }
-    else static if (isPointer!S && is(S : const(char)*))
+    else static if (isPointer!S && (is(S : const(char)*) || is(S : const(wchar)*) || is(S : const(dchar)*)))
     {
-        import core.stdc.string : strlen;
-        // It is unsafe because we cannot guarantee that the pointer is null terminated.
-        return value ? cast(T) value[0 .. strlen(value)].dup : null;
+        // This is unsafe because we cannot guarantee that the pointer is null terminated.
```

Comment:
in phobos there is an idioms of directly call @trusted delegates. you could make a @system delegate in place.
if you look at the error message currently generated you still have to look into "to" to find out what is not safe or trusted. And for that you have to understand the code. Which means work. But if you see "@system" you don't have to understand the code anymore